http_requests.py

```diff
@@ -221,3 +226,66 @@ def delete(self, url: str, return_type=None, **kwargs):
                 return_type=None,
                 **kwargs
         )
+
+
+class Encoder(enum.StrEnum):
+    GZIP = 'gzip'
+
+
+def encode_request(
+    data: str | bytes | dict | typing.IO=None,
+    json: dict=None,
+    headers: dict[str, str]=None,
+    encoder: Encoder=Encoder.GZIP,
+) -> (
+    tuple[collections.abc.Generator[bytes, None, None], dict[str, str]]
+    | collections.abc.Generator[bytes, None, None]
+):
+    '''
+    Encodes the given `data` or `json` property based on the selected `encoder`. If both `data` and
+    `json` are provided, the `data` property is used (same behaviour as implemented by the
+    `requests` library).
+
+    The corresponding `Content-Encoding` header is patched-in to the provided headers dictionary.
+
+    If `headers` is provided and not `None`, the response is a tuple of the compression result and
+    the patched headers, otherwise only the compression result is returned.
+    '''
+    if not data and json is not None:
+        data = js.dumps(json)
+
+    def _encode(obj, encoding: str='utf-8') -> bytes:
+        if isinstance(obj, bytes):
+            return obj
+        elif isinstance(obj, str):
+            return obj.encode(encoding=encoding)
+        elif isinstance(obj, dict):
+            return js.dumps(obj).encode(encoding=encoding)
+        else:
+            raise NotImplementedError(f'Encoding of type {type(obj)} is not (yet) supported')
+
+    def _compress(data, encoder) -> collections.abc.Generator[bytes, None, None]:
```

Comment:
nit: `encoding_method` rather than `encoder` as this is not used directly but as a compressor mapping